File: zalagaonica-web/src/types/incomingDocument.ts
// src/types/incomingDocument.ts

export interface IncomingDocument {
  id: number;
  supplierName: string;
  bookingDate: string; // Datum knjiženja
  documentNumber: string; // Ulazni dok.
  documentDate: string; // Datum
  purchaseValue: number; // Nabavna vrijednost
  margin: number; // Marža
  tax: number; // Porez
  status: 'otvoren' | 'proknjižen';
  warehouseName: string; // Naziv skl.
  documentType: string; // Tip dokumenta
  year: number;
  operator: string;
  dueDate: string;
  isPosted: boolean;
  invoiceValue: number;
  discount: number;
  cost: number;
  wholesaleValue: number;
  vatAmount: number;
  retailValue: number;
  returnFee: number;
  totalWithReturnFee: number;
  pretaxAmount: number;
  totalPaid: number;
  note?: string;
}

File: backend/Zalagaonica.Backend/Domain/Entities/IncomingDocument.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class IncomingDocument
    {
        [Key]
        public Guid Id { get; set; }

        [Required, MaxLength(100)]
        public string DocumentNumber { get; set; } = string.Empty;

        [MaxLength(200)]
        public string? SupplierName { get; set; }

        public DateTime DateReceived { get; set; } = DateTime.UtcNow;

        [MaxLength(300)]
        public string? Description { get; set; }

        // Extended fields to match frontend requirements
        public DateTime BookingDate { get; set; } = DateTime.UtcNow;

        public DateTime DocumentDate { get; set; } = DateTime.UtcNow;

        [Range(-999999999, 999999999)]
        public decimal PurchaseValue { get; set; }

        [Range(-999999999, 999999999)]
        public decimal Margin { get; set; }

        [Range(0, 100)]
        public decimal Tax { get; set; }

        [MaxLength(50)]
        public string Status { get; set; } = "otvoren"; // 'otvoren' or 'proknjižen'

        [MaxLength(200)]
        public string? WarehouseName { get; set; }

        [MaxLength(50)]
        public string DocumentType { get; set; } = "PRIMKA";

        public int Year { get; set; }

        [MaxLength(100)]
        public string? Operator { get; set; }

        public DateTime? DueDate { get; set; }

        public bool IsPosted { get; set; } = false;

        public decimal InvoiceValue { get; set; }

        [Range(0, 100)]
        public decimal Discount { get; set; }

        public decimal Cost { get; set; }

        public decimal WholesaleValue { get; set; }

        public decimal VatAmount { get; set; }

        public decimal RetailValue { get; set; }

        public decimal ReturnFee { get; set; }

        public decimal TotalWithReturnFee { get; set; }

        public decimal PretaxAmount { get; set; }

        public decimal TotalPaid { get; set; }

        [MaxLength(500)]
        public string? Note { get; set; }
    }
}
